src/dgram.cpp

```diff
@@ -0,0 +1,321 @@
+/*
+    Copyright (c) 2007-2016 Contributors as noted in the AUTHORS file
```

Comment:
Copyright range is a bit wrong :-) Better to these legal things right the first time